internal/xds/server/kubejwt/tokenreview.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package kubejwt
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	authenticationv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// GetKubernetesClient creates a Kubernetes client using in-cluster configuration.
+func GetKubernetesClient() (*kubernetes.Clientset, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create in-cluster config: %w", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Kubernetes client: %w", err)
+	}
+
+	return clientset, nil
+}
+
+func validateKubeJWT(ctx context.Context, clientset *kubernetes.Clientset, token string) (bool, error) {
+	tokenReview := &authenticationv1.TokenReview{
```

Comment:
are we validating that these token belongs to an envoy created by EG ?

Reply:
Just by the service account, I was thinking about it, we will need to have an exact name of the envoy pod

Reply:
Can we leverage the IR keys in the snapshot cache which map to the fleet deployment name and namespace?

Reply:
We can use `kubernetes.io.namespace` and `kubernetes.io.name` claims to verify the pod.

https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#schema-for-service-account-private-claims

Reply:
Tracked in this issue: https://github.com/envoyproxy/gateway/issues/5863

Reply:
I have added the basic check for deployment in snapshot cache, we will need to improve it and make it work with merged gateways mode